clients/unshaded/src/main/java/tachyon/client/lineage/LineageContext.java

```diff
@@ -53,4 +53,15 @@ public LineageMasterClient acquireMasterClient() {
   public void releaseMasterClient(LineageMasterClient masterClient) {
     mLineageMasterClientPool.release(masterClient);
   }
+
+  /**
+   * Re-initializes the {@link LineageContext}. This method should only be used in
+   * {@link ClientContext}.
+   */
+  public void reset() {
+    if (mLineageMasterClientPool != null) {
```

Comment:
I think it's fine for it to be unsynchronized since reset is a temporary hack to allow tests to reset the configuration. This method should never be called while there are multiple threads trying to access the `LineageContext`. I've created https://tachyon.atlassian.net/browse/TACHYON-1623 to address the issue.